include/std/experimental/conqueue

```diff
@@ -440,6 +471,18 @@ struct buffer_queue<T, Alloc>::pop_sender {
             "async_pop: queue is empty, putting %p in the waiters queue\n",
             &op);
         self.waiters.push_back(&op);
+        // We are in the queue, set up the stop callback.
+        if constexpr (stoppable) {
+          op.stop_callback.emplace(stop_token, cancel_callback{op});
+          if (--op.stop_count == 0) {
+            // cancel callback has already been called, observed that we are in
+            // the middle of construction and left cancellation up to us.
+            self.waiters.remove(&op);
+            lock.unlock();
```

Comment:
I presume you are holding the lock across the registration of the callback to avoid the need to handle some other thread potentially completing this op concurrently with the registration of the stop-callback.

If you are going to hold the mutex across the registration of the stop-callback, you could avoid adding and then just having to remove the queue item again later and only add it once you're sure you won't be immediately completing.

```
if constexpr (stoppable) {
  op.stop_callback.emplace(stop_token, cancel_callback{op});
  if (--op.stop_count == 0) {
    lock.unlock();
    op.stop_callback.reset();
    stdexec::set_stopped((Receiver&&)op.receiver);
  } else {
    self.waiters.push_back(&op);
  }
} else {
  self.waiters.push_back(&op);
}
// Just let implicit lock.unlock() be called on scope exit here.
```